### DISCUSSION TITLE
How to probe Z?

### DISCUSSION THREAD

**Discussion by stevempotter** [terjeio/ioSender#125] [Q&A]
OK, I have gotten my MPCNC machine to move for the first time using IOsender with a .gcode file from VcarvePro. Exciting!  But I can't figure out how to use the Z probe. Is there a wiki or guide on how to use the Probe tab? 

I expected there to be a button that I hit and then it moves down until it hits the plate (I have set the touchplate thickness = 3mm) and then it sets Z-zero to 3mm below that Z coordinate, and moves back up 10 mm. I can't even find the Go button. I don't understand about coordinate systems and offsets.  I wish there were some more documentation for newbies like me learning to use IOsender. Maybe there is and I just can't find it?

**Comment by phil-barrett** [comment]
Have you looked at the Probing tab?  Edge finder, external, check probe Z and click in the center of the diagram.  

[edit] you probably want to use the Set offset Action rather than Set coordinate system

By the way, I think the "NOTE: only lightly tested" text can be removed at this point.
![image](https://user-images.githubusercontent.com/16570958/127747382-2c0ef04b-0040-4eb8-ba58-b685a6994754.png)

**Reply by stevempotter** [reply]
Thanks, that worked great for setting Z. Is there a tutorial somewhere on how to use all these probing features and what kind of touchplate one needs to make them work when carving wood? (such as, how to correct for a skewed position on the table, or a warped piece of wood?)

**Reply by terjeio** [reply]
>  Is there a tutorial somewhere...

Not that I know of except [this](https://vers.by/en/blog/useful-articles/probe-screen) which I have used as a template for my implementation. It would be great if somebody could contribute by writing one for the sender, even better with links to videos?

**Reply by raenji-sk** [reply]
Got some videos on my YT channel where I used the probing screens. Z, center finder and also rotation.
https://youtu.be/DoD-knKYlZk
Starts at around 10:30 for probing. Rotation is in another video but I did not go into much detail.
I changed the controller to Uccnc so unfortunately can't help any further.
